controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentMetrics.java

```diff
@@ -14,24 +15,30 @@
  */
 public class DeploymentMetrics {
 
-    public static final DeploymentMetrics none = new DeploymentMetrics(0, 0, 0, 0, 0, Optional.empty());
+    public static final DeploymentMetrics none = new DeploymentMetrics(0, 0, 0, 0, 0, Optional.empty(), Map.of());
 
     private final double queriesPerSecond;
     private final double writesPerSecond;
     private final double documentCount;
     private final double queryLatencyMillis;
     private final double writeLatencyMills;
     private final Optional<Instant> instant;
+    private final Map<Warning, Integer> warnings;
 
     /* DO NOT USE. Public for serialization purposes */
     public DeploymentMetrics(double queriesPerSecond, double writesPerSecond, double documentCount,
-                             double queryLatencyMillis, double writeLatencyMills, Optional<Instant> instant) {
+                             double queryLatencyMillis, double writeLatencyMills, Optional<Instant> instant,
+                             Map<Warning, Integer> warnings) {
         this.queriesPerSecond = queriesPerSecond;
         this.writesPerSecond = writesPerSecond;
         this.documentCount = documentCount;
         this.queryLatencyMillis = queryLatencyMillis;
         this.writeLatencyMills = writeLatencyMills;
         this.instant = Objects.requireNonNull(instant, "instant must be non-null");
+        this.warnings = Map.copyOf(Objects.requireNonNull(warnings, "warnings must be non-null"));
+        if (warnings.entrySet().stream().anyMatch(kv -> kv.getValue() < 0)) {
```

Comment:
Won't each entry always be >0, so you can check for `<=0`? But is the check necessary at all? Is it because of serialization? If it's kept, there should also be a unit test for this case.

Reply:
It's just a part of the contract of this class. Negative counts never make sense so they are rejected, but there's not really any reason to reject 0 at this level.

Reply:
Ok, I see it now.